### PR TITLE
(DOCSP-26996) Add Realm React support to 'Write Transactions'

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

I took this opportunity to bring this page's content in line with the Flutter SDK, as the existing content is hard to follow and overly explanatory.

### Jira

- https://jira.mongodb.org/browse/DOCSP-26996

### Staged Changes

- [Write Transactions](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-26996/sdk/react-native/realm-database/write-transactions/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/realm-database/write-transactions.txt [hunk 1]
```diff
@@ -11,41 +11,18 @@ Write Transactions - React Native SDK
...
-          year: 2018,
-          miles: 30000
-        })
-      })
+.. literalinclude:: /examples/generated/react-native/ts/create-test.snippet.crud-create-object.tsx
+  :language: typescript
+  :emphasize-lines: 6-8
 
 Transaction Lifecycle
 ---------------------
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/write-transactions.txt:28]: nit while we're mucking around in here. below, rather than having the note about how transactions work w sync, could you make that either:

- another bullet point
- a sub bullet point
- or just another sentence on this bullet point.

File: source/sdk/react-native/realm-database/write-transactions.txt [hunk 2]
```diff
@@ -11,41 +11,18 @@ Write Transactions - React Native SDK
    :depth: 2
    :class: singlecol
 
-A write transaction is a function that modifies objects in a realm. Write 
-transactions let you create, modify, or delete Realm objects. They handle 
-operations in a single, idempotent update. A transaction is 
-*all or nothing*. Either:
+Within a ``<RealmProvider>`` that has an open realm, you can create Realm objects
+using a :js-sdk:`Realm.write() transaction block <Realm.html#write>`.
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/write-transactions.txt:15]:
```suggestion
using a :js-sdk:`Realm.write() <Realm.html#write>` transaction block.
```

usually we just link to the method name.

**Review by mongodben — APPROVED** [review]
lgtm w a few small suggestions.

File: source/sdk/react-native/realm-database/write-transactions.txt [hunk 3]
```diff
@@ -11,41 +11,18 @@ Write Transactions - React Native SDK
    :depth: 2
    :class: singlecol
 
-A write transaction is a function that modifies objects in a realm. Write 
-transactions let you create, modify, or delete Realm objects. They handle 
-operations in a single, idempotent update. A transaction is 
-*all or nothing*. Either:
+Within a ``<RealmProvider>`` that has an open realm, you can create Realm objects
```
Comment (mongodben) [inline source/sdk/react-native/realm-database/write-transactions.txt:14]:
```suggestion
Within a ``<RealmProvider>`` that has an open realm, access a realm
with the ``useRealm()`` hook. You can create, update, and delete Realm objects
```

File: source/sdk/react-native/realm-database/write-transactions.txt [hunk 4]
```diff
@@ -11,41 +11,19 @@ Write Transactions - React Native SDK
    :depth: 2
    :class: singlecol
 
-A write transaction is a function that modifies objects in a realm. Write 
-transactions let you create, modify, or delete Realm objects. They handle 
-operations in a single, idempotent update. A transaction is 
-*all or nothing*. Either:
+Within a ``<RealmProvider>`` that has an open realm, you can access a realm with
```
Comment (takameyer) [inline source/sdk/react-native/realm-database/write-transactions.txt:14]:
```suggestion
Within a ``<RealmProvider>``, you can access a realm with
```
Anything rendered within a RealmProvider will have access to an open realm.  If it's closed, then it won't be rendered.  Therefore I think we can remove "that has an open realm", as this may lead to confusion.

Comment (takameyer) [reply source/sdk/react-native/realm-database/write-transactions.txt:14]: Also we should capitalize "Realm"

File: source/sdk/react-native/realm-database/write-transactions.txt [hunk 5]
```diff
@@ -61,12 +39,10 @@ After processing a transaction, Realm Database either **commits** it or
   Once applied, the realm automatically updates :ref:`live queries
   <react-native-live-queries>`. It notifies listeners of created, modified, 
   and deleted objects.
-  
-  .. note::
      
-     When using :ref:`Sync <sync>`, the SDK also queues
-     the changes to send to Atlas App Services. The SDK sends
-     these changes when a network is available.
+  - When using :ref:`Sync <sync>`, the SDK also queues
```
Comment (takameyer) [inline source/sdk/react-native/realm-database/write-transactions.txt:43]: Might be worth adding a bullet that after a write transaction, any object or collections returned by `useQuery` or `useObject` will rerender for any relevant changes.

**Review by takameyer — APPROVED** [review]
Just a few small things.  Otherwise looks good!

File: source/sdk/react-native/realm-database/write-transactions.txt (hunk 4 above)
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/write-transactions.txt:14]: Good point! Thanks, Andrew. 

Re: capitalization, we typically only capitalize "Realm" when talking about the product itself. An instance of a realm is lower case. I'm not sure the history of this practice, but I imagine it's to improve readability. We use the latter form a lot in the docs.

File: source/sdk/react-native/realm-database/write-transactions.txt (hunk 5 above)
Comment (krollins-mdb) [reply source/sdk/react-native/realm-database/write-transactions.txt:43]: Ah, that's excellent information to add. Thank you!

File: source/sdk/react-native/realm-database/write-transactions.txt (hunk 4 above)
Comment (takameyer) [reply source/sdk/react-native/realm-database/write-transactions.txt]: > capitalization

Yeah, that sounds logical.  As long as you all have a way to keep it consistant throughout the docs, i'm happy 👍🏼
